source/developer_manual/design/processes.rst

```diff
@@ -0,0 +1,146 @@
+.. _dovecot_processes:
+
+=================
+Dovecot processes
+=================
+
+Dovecot is split into multiple processes where each process does only
+one thing. This is partially because it makes the code cleaner, but also
+because it allows setting up different privileges for each process. The
+most important processes are:
+
+-  Master process (dovecot)
+-  Login processes (imap-login, pop3-login)
+-  Authentication process (dovecot-auth)
+-  Mail processes (imap, pop3)
+
+
+Master process
+--------------
+
+This process keeps all the other processes running. If a child process
+dies, another one is restarted automatically. It always runs as root,
+unless you're specifically running everything under a single normal UID.
+
+The master process reads the configuration file and exports the settings
+to other processes via environment variables.
+
+All logging also goes through master process. This avoids problems with
+rotating log files, as there's only a single process to send a signal to
+reopen the log file. Also writing to the same log file (if not using
+syslog) isn't necessarily safe to do in multiple processes concurrently.
+
+Making the logging go through master process also gives a couple of
+advantages from security and reliability point of view: All log lines
+can be prefixed with the process's name and the username of the user who
+was logged in, without the possibility for the process itself to forge
+them. Flooding logs can also be prevented. By default Dovecot allows
+non-privileged processes to write 10 lines per second before it begins
+to delay reading their input, which finally causes the badly behaving
+process to start blocking on writing to stderr instead of eating all the
+CPU and disk space.
+
+In the Dovecot 2.0 design, the master process is split to three parts:
+the Master process which does nothing more than keep the processes
+running, the config process which handles reading the configuration file
+(supporting also eg. SQL storages!) and the log process which handles
+the logging.
```

Comment:
Ah.

We shouldn't start with describing archaic behavior and then mention what is currently really going on. We should drop descriptions for old stuff entirely or at least move it to some unimportant footnote.